### PR TITLE
add stimSource stimTarget and synMech

### PR DESCRIPTION
**Pull request by rodriguez-facundo**
Added fields to stemSource stimTarget and synMech

### CODE REVIEW

File: components/definition/stimulation/NetPyNEStimulationTarget.js [hunk 1]
```diff
@@ -1,29 +1,60 @@
...
+import Conds from 'material-ui/svg-icons/maps/local-offer';
+import StimTarget    from 'material-ui/svg-icons/action/reorder';
+
+const CondsIcon = <Conds/>;
+const StimIcon  = <StimTarget />;
+ 
 var PythonControlledCapability = require('../../../../../js/communication/geppettoJupyter/PythonControlledCapability');
 var PythonControlledTextField = PythonControlledCapability.createPythonControlledControl(TextField);
+var PythonControlledListComponent = PythonControlledCapability.createPythonControlledControl(ListComponent);
+var PythonMethodControlledSelectField = PythonControlledCapability.createPythonControlledControlWithPythonDataFetch(SelectField);
```
Comment (adrianq) [inline components/definition/stimulation/NetPyNEStimulationTarget.js]: @rodriguez-facundo Is this used anywhere?

File: components/definition/stimulation/NetPyNEStimulationSource.js [hunk 2]
```diff
@@ -10,79 +10,259 @@ import RaisedButton from 'material-ui/RaisedButton';
...
     var that = this;
     var storedValue = this.props.name;
     var newValue = event.target.value;
     this.setState({ currentName: newValue });
     this.triggerUpdate(function () {
       // Rename the population in Python
-      Utils.renameKey('netParams.stimSourceParams', storedValue, newValue, (response, newValue) => { that.renaming=false;});
-      that.renaming=true;
+      Utils.renameKey('netParams.stimSourceParams', storedValue, newValue, (response, newValue) => { that.renaming = false });
+      that.renaming = true;
```
Comment (adrianq) [inline components/definition/stimulation/NetPyNEStimulationSource.js]: Do we actually need 'that'? What's the value of 'this' inside that method?

File: components/definition/stimulation/NetPyNEStimulationSource.js [hunk 3]
```diff
@@ -10,79 +10,259 @@ import RaisedButton from 'material-ui/RaisedButton';
...
+              model={"netParams.stimSourceParams['" + this.props.name + "']['i']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.vc">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['vc']"}
+            />
+          </NetPyNEField>
+        </div>
+    } else {
```
[the remark this answers is not in the document]
Comment (adrianq) [reply components/definition/stimulation/NetPyNEStimulationSource.js]: An any other else of this kind in the following lines...

File: components/definition/stimulation/NetPyNEStimulationSource.js [hunk 4]
```diff
@@ -10,79 +10,259 @@ import RaisedButton from 'material-ui/RaisedButton';
...
+        </div>
+    } else if (this.state.sourceType=='SEClamp'){
+      var variableContent = 
+        <div>
+          <NetPyNEField id="netParams.stimSourceParams.rs">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['rs']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.dur" className="listStyle">
```
Comment (adrianq) [inline components/definition/stimulation/NetPyNEStimulationSource.js]: Same id for two different fields

File: components/definition/stimulation/NetPyNEStimulationTarget.js (hunk 1 above)
Comment (rodriguez-facundo) [reply components/definition/stimulation/NetPyNEStimulationTarget.js]: the first one is used to add multiple sections and the second one is used to select the stimSource that will be used from the list of stimSources previously created.

File: components/definition/stimulation/NetPyNEStimulationSource.js (hunk 2 above)
Comment (rodriguez-facundo) [reply components/definition/stimulation/NetPyNEStimulationSource.js]: I saw in NetPyNePopulation.js that code to handle population rename, so I used the same one. Maybe we could do it differently.

File: components/definition/stimulation/NetPyNEStimulationTarget.js (hunk 1 above)
Comment (adrianq) [reply components/definition/stimulation/NetPyNEStimulationTarget.js]: You are right, my bad!

File: components/definition/stimulation/NetPyNEStimulationSource.js (hunk 4 above)
Comment (adrianq) [reply components/definition/stimulation/NetPyNEStimulationSource.js]: @rodriguez-facundo I saw that you changed the id for all the components in the file. I just meant two components with the same id "netParams.stimSourceParams.dur". Probably it is better if you revert that change. metadata.py uses the ids to extract the help, hint, type, etc.

**Comment by adrianq** [comment]
@rodriguez-facundo I have just merged this PR but there were actually a few pending issues. I have implemented them so we don't delay this PR anymore:
- I was getting a 'variable has not been defined' js error. I fix it here (https://github.com/MetaCell/geppetto-netpyne/pull/10/commits/2c9d04018d00e4ca7c4b6e431a45e4ddb910e159)
- As I said in my previous comment, stimTargetParams.sec is not a list but a string (https://github.com/MetaCell/geppetto-netpyne/pull/10/commits/d4f9a45188d1a9745bdbe80bfebb6126b3681dda). I also modified the metadata in netpyne
- Deleted some debugging logs you added (https://github.com/MetaCell/geppetto-netpyne/pull/10/commits/25d8c8c120896ef24095ca87dbd12c535e5307ec)
- I have created a card to address the duplicity between SimulationRange and Range (https://trello.com/c/RlWxqqnc/197-merge-simulationrange-and-range-component)

**Pull request merged**